cmd/copyUtil.go

```diff
@@ -45,23 +45,31 @@ type copyHandlerUtil struct{}
 var gCopyUtil = copyHandlerUtil{}
 
 // checks if a given url points to a container or virtual directory, as opposed to a blob or prefix match
-func (util copyHandlerUtil) urlIsContainerOrVirtualDirectory(url *url.URL) bool {
-	if azblob.NewBlobURLParts(*url).IPEndpointStyleInfo.AccountName == "" {
+func (util copyHandlerUtil) urlIsContainerOrVirtualDirectory(rawURL string) bool {
+	parsedURL, err := url.Parse(rawURL)
```

Comment:
This url parse is done here mostly to make the Path calls below easier. I could remove it and get the path by appending parts.Container name and parts.Blob name but it might get tricky since we also support IP endpoint style, which means the account name is also part of the path.